text_extensions_for_pandas/io/watson/nlu.py

```diff
@@ -337,14 +365,65 @@ def _make_relations_dataframe_zero_copy(relations):
     return table.to_pandas()
 
 
+def _make_entity_mentions_dataframe(entities: List,
+                                    original_text: str,
+                                    apply_standard_schema: bool) -> pd.DataFrame:
+    """
+    Unroll the records of the "mentions" element of NLU entities into a flat
+    DataFrame. Schema of this DataFrame is `_entity_mentions_schema`
+    above.
+    :param entities: The "entities" section of a parsed NLU response
+    :param original_text: Text of the document.  This argument must be provided if there
+     are entity mention spans.
+    :param apply_standard_schema: Value of the eponymous argument from `parse_response`.
+    """
+    if 0 == len(entities) or "mentions" not in entities[0].keys():
+        # No mentions to unroll. Return an empty DataFrame.
+        return util.apply_schema(
+            pd.DataFrame(columns=[e[0] for e in _entity_mentions_schema]),
+            _entity_mentions_schema, apply_standard_schema)
+    if original_text is None:
+        raise ValueError(
+            "Unable to construct target text for converting entity mentions to spans")
+    # Explode out the nested relations containing entity location information.
+    # If there was a version of DataFrame.explode() that could handle structs,
+    # we would be able to vectorize this operation.
```

Comment:
Could you use `util.make_table` which flattens the struct columns? That is pretty much the reason I made that function so that you could work with non-nested columns in a vectorized way after flattening.

Reply:
I tried to, but it wasn't working for me. How about if we merge this current change set, and then you can take a look at your leisure and see if you can remove that loop from the Python code?

Reply:
sure, will do